src/ORM/Association/HasMany.php

```diff
@@ -134,6 +181,67 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
     }
 
     /**
+     * Deletes/sets null the related objects according to the dependency between source and targets and foreign key nullability
+     * Skips deleting records present in $remainingEntities
+     *
+     * @param array $properties array of foreignKey properties
+     * @param EntityInterface $entity the entity which should have its associated entities unassigned
+     * @param Table $target The associated table
+     * @param array $remainingEntities Entities that should not be deleted
+     * @return void
+     */
+    protected function _unlinkAssociated(array $properties, EntityInterface $entity, Table $target, array $remainingEntities)
```

Comment:
The records are being added fine, the problem is the unlinking of existing records, as these statements will cause nothing to be deleted, since arithmetic comparisons against `NULL` are always `NULL` (at least that is true for MySQL and SQLite, not sure about Postgre and SQL Server). Comparing against `NULL` must use `IS NULL` or `IS NOT NULL`. However in this case these conditions should not be present at all, they should be filtered out.